query-connector/src/app/query/components/stepIndicator/TitleBox.tsx

```diff
@@ -0,0 +1,31 @@
+import { Mode } from "@/app/constants";
+import { PAGE_TITLES } from "./StepIndicator";
+
+type TitleBoxProps = {
+  step: Mode;
+};
+
+/**
+ * TitleBob that renders the page title and step counter state
```

Comment:
Good idea to move this into it's own component! Small typo here in the component docs